rescheduler/rescheduler.go

```diff
@@ -458,17 +458,39 @@ func groupPods(client kube_client.Interface, node *apiv1.Node) ([]*apiv1.Pod, []
 	return requiredPods, otherPods, nil
 }
 
-func filterCriticalPods(allPods []*apiv1.Pod, podsBeingProcessed *podSet) []*apiv1.Pod {
-	criticalPods := []*apiv1.Pod{}
+func filterCriticalPodsCreatedByDaemonSets(allPods []*v1.Pod, podsBeingProcessed *podSet) []*v1.Pod {
+	criticalPods := []*v1.Pod{}
 	for _, pod := range allPods {
-		if isCriticalPod(pod) && !podsBeingProcessed.Has(pod) {
+		if isCriticalPod(pod) && isDaemonsetPod(pod) && !podsBeingProcessed.Has(pod) {
 			criticalPods = append(criticalPods, pod)
 		}
 	}
 	return criticalPods
 }
 
-func isCriticalPod(pod *apiv1.Pod) bool {
+func isCriticalPod(pod *v1.Pod) bool {
+	// Shouldn't we check "kube-system" namespace as well?
 	_, found := pod.ObjectMeta.Annotations[criticalPodAnnotation]
+	criticalBasedOnPriority := false
```

Comment:
If `found` is true, you can return early:

```go
func isCriticalPod(pod *v1.Pod) bool {
	if _, found := pod.ObjectMeta.Annotations[criticalPodAnnotation]; found {
		return true
	}
	return pod.Spec.Priority != nil && *pod.Spec.Priority >= SystemCriticalPriority
}
```

Reply:
Yes, thanks. I will make that change.